### PR TITLE
Redis

### PR DESCRIPTION
**Pull request by ajax-butriy-v**

### CODE REVIEW

File: pass-manager-svc/src/main/kotlin/com/example/passmanagersvc/repositories/impl/RedisPassOwnerRepository.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    override fun findById(passOwnerId: String): Mono<MongoPassOwner> {
+        val key = passOwnerKey(passOwnerId)
+        return reactiveRedisTemplate.opsForValue().get(key)
+            .handle { item, sink ->
+                if (item.isEmpty()) {
+                    sink.error(PassOwnerNotFoundException("Could not find pass owner by id $passOwnerId"))
+                } else {
+                    sink.next(item)
+                }
+            }
```
Comment (ajax-malov-n) [inline pass-manager-svc/src/main/kotlin/com/example/passmanagersvc/repositories/impl/RedisPassOwnerRepository.kt:37]: Maybe extract this

File: pass-manager-svc/src/main/kotlin/com/example/passmanagersvc/configuration/RedisConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import java.time.Duration
+
+@Configuration
+class RedisConfiguration(
+    @Value("\${spring.data.redis.timeout.millis}")
+    private val redisCommandTimeoutInMillis: Long,
+    @Value("\${spring.data.redis.port}")
+    private val redisPort: Int,
+    @Value("\${spring.data.redis.host}")
+    private val redisHost: String,
```
Comment (ajax-shulepa-a) [inline pass-manager-svc/src/main/kotlin/com/example/passmanagersvc/configuration/RedisConfiguration.kt]: maybe use ConfugurationProperties?

File: pass-manager-svc/src/main/kotlin/com/example/passmanagersvc/service/impl/PassManagementServiceImpl.kt [hunk 3]
```diff
@@ -14,6 +15,7 @@ import reactor.kotlin.core.util.function.component2
 internal class PassManagementServiceImpl(
     private val passOwnerService: PassOwnerService,
     private val passService: PassService,
+    @Qualifier("mongoPassRepository")
```
Comment (ajax-shulepa-a) [inline pass-manager-svc/src/main/kotlin/com/example/passmanagersvc/service/impl/PassManagementServiceImpl.kt]: is it ok that service should define exact implementation of repository?

File: pass-manager-svc/src/main/kotlin/com/example/passmanagersvc/repositories/impl/RedisPassOwnerRepository.kt [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            .map { objectMapper.readValue<MongoPassOwner>(it) }
+            .switchIfEmpty {
+                mongoPassOwnerRepository.findById(passOwnerId)
+                    .flatMap(::savePassOwnerToRedis)
+                    .switchIfEmpty {
+                        reactiveRedisTemplate.opsForValue()
+                            .set(key, byteArrayOf(), Duration.ofMinutes(redisExpirationTimeoutInMinutes))
+                            .then(Mono.empty())
+                    }
```
Comment (ajax-shulepa-a) [inline pass-manager-svc/src/main/kotlin/com/example/passmanagersvc/repositories/impl/RedisPassOwnerRepository.kt]: extract to separate function

File: pass-manager-svc/src/main/kotlin/com/example/passmanagersvc/repositories/impl/RedisPassOwnerRepository.kt [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        val byteArray = objectMapper.writeValueAsBytes(passOwner)
+        return reactiveRedisTemplate.opsForValue().set(
+            key,
+            byteArray,
+            Duration.ofMinutes(redisExpirationTimeoutInMinutes)
+        ).thenReturn(passOwner)
+    }
+
+    companion object {
+        private const val KEY_PREFIX = "key-"
```
Comment (ajax-shulepa-a) [inline pass-manager-svc/src/main/kotlin/com/example/passmanagersvc/repositories/impl/RedisPassOwnerRepository.kt]: what if you save more entities together with the pass owner in redis? even If no collision by id? how will you differentiate entries from each other?

**Review by ajax-shulepa-a — APPROVED** [review]

**Pull request merged**